qiskit/quantum_info/synthesis/clifford_decompose.py

```diff
@@ -488,11 +491,13 @@ def decompose_clifford_greedy(clifford):
     while len(qubit_list) > 0:
         # Calculate the adjoint of clifford_cpy without the phase
         clifford_adj = clifford_cpy.copy()
-        tmp = clifford_adj.destabilizer.X.copy()
-        clifford_adj.destabilizer.X = clifford_adj.stabilizer.Z.T
-        clifford_adj.destabilizer.Z = clifford_adj.destabilizer.Z.T
-        clifford_adj.stabilizer.X = clifford_adj.stabilizer.X.T
-        clifford_adj.stabilizer.Z = tmp.T
+        clifford_adj.paulis._phase -= clifford_adj.paulis._count_y()
+        tmp = clifford_adj.destabilizers().x.copy()
+        clifford_adj.destabilizers().x = clifford_adj.stabilizers().z.T
+        clifford_adj.destabilizers().z = clifford_adj.destabilizers().z.T
+        clifford_adj.stabilizers().x = clifford_adj.stabilizers().x.T
+        clifford_adj.stabilizers().z = tmp.T
+        clifford_adj.paulis._phase -= clifford_adj.paulis._count_y()
```

Comment:
If we need to be able to set the stabilisers and destabilisers separately, then we need to revisit Shelly's comment (https://github.com/Qiskit/qiskit-terra/pull/7269/files#r811744330), because there's nothing in the docstring/contract/tests of the `Clifford.stabilizers()` method that ensure that you can assign to the output.  It doesn't seem like good form to deprecate this without a proper ergonomic alternative.

In this particular case, I feel like logically the intent is simply to construct an entirely new object.  I think we can keep the current `stabilizers()` form as-is if we instead a constructor classmethod `Clifford.from_stabilizers` or something, which is possibly better than adding setters (it's more memory efficient, for one).

Reply:
I see. Then I would write it in the form of assigning to paulis attribute if it involves a setter.
(Clifford cannot be made from only stabilizer, Clifford needs both stabilizer and destabilizer.)

Reply:
> Clifford cannot be made from only stabilizer, Clifford needs both stabilizer and destabilizer

Yeah, sorry if I wasn't clear, I meant a method that looked like
```python
class Clifford:
    @classmethod
    def from_stabilizers(cls, stabilizers, destabilizers):
        ...
```
(so it had both).  I don't know if there's a better name.

Reply:
Basically, the paulis attribute setter is sufficient. However, the method might be useful :+1:

Reply:
Done in 538f2e1